src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -221,14 +221,22 @@ trait Implicits {
     }
 
     def isCyclicOrErroneous: Boolean = {
-      if (!isCyclicOrErroneousCache.isKnown) isCyclicOrErroneousCache = computeIsCyclicOrErroneous
-      isCyclicOrErroneousCache.booleanValue
+      if (!isCyclicOrErroneousCache.isKnown)
+        try global.withPropagateCyclicReferences {
+          val cyclicOrErroneous = sym.hasFlag(LOCKED) || containsError(tpe)
+          isCyclicOrErroneousCache = cyclicOrErroneous
+          cyclicOrErroneous
+        } catch {
+          case _: CyclicReference =>
```

Comment:
Hm. `validate-test` took somewhat longer on this commit than the previous... some others take longer... I can't tell if that's in normal variation or whatnot.